### DISCUSSION TITLE
What does 'publication ready' mean?

### DISCUSSION THREAD

**Discussion by teunbrand** [ggplot2-extenders/ggplot-extension-club#65] [Q&A]
I [started asking](https://bsky.app/profile/teunbrand.bsky.social/post/3lbwkxoopgk2b) what it means for plots to be 'publication ready' and I didn't really learn what I wanted. 

I always scratch my head when this term pops up because I can't really quite lay a finger on what it means. People have published raw, all-default ggplot2 figures in academic journals, so while it could refer to some level of polish on a plot, it doesn't need to. On the other hand, some non-academic visualisations have incredible levels of polish. These are never intended to be published in an academic journal so they can take on things like a bunch of titles, text labels and other annotations that are almost never used in academic journals. So, despite their polish, these don't qualify as 'publication ready' I think.

Do y'all have any opinions or can describe qualities of work that falls under 'publication ready' versus not?

**Comment by yjunechoe** [comment]
It's completely vacuous to me - I lowkey don't like the marketing that accompanies the term "publication ready". But you're right to note that there are (often unspoken) rules to academic figures, such that a journal wouldn't publish a manuscript with a typical WSJ or NYT style data journalism graph with lots of custom annotations and highlighting (at best distracting and frivolous, at worst leading/biasing the readers to the authors' favored conclusions about the data).

So maybe "publication ready" could (should?) simply refer to a kind of an aesthetic style that means "looks pretty and polished while abiding by the constraints of academic professionalism". I don't think it reflects reality to think of "publication ready" as processual, contrasted with "not-yet-ready", unless you're publishing to Nature or Scientific American or similar (the kind of journals where layfolk science enthusiasts would subscript to for monthly hard copies), which cares about polishedness and encourages data journalism style graphs (and so may strike down a default theme ggplot as not ready for publication).

Edit: not a critique of tidyplot -- I didn't realize that was the original context of this thread 😅

**Reply by teunbrand** [reply]
I'm inclined to agree that there is a marketing aspect to the term that sits unpleasantly in the mind. The 'abiding by the constraints of academic professionalism' is an interesting piece and I guess it throws out all sorts of misleading techniques (e.g. bar charts that don't start at 0, spurious correlations through meddling with dual y-axes etc.).

**Reply by yjunechoe** [reply]
Not speaking to tidyplot specifically (I still haven't really looked at what it does), but speaking from my own experience the "unpleasantly" is especially strong after seeing predatory tactics along those lines used by a special breed of self-help gurus who target early-career academics and feed off their imposter syndrome. I love data viz but it annoys me when I see people act holier-than-thou and generate FOMO over the latest fad in styling your graph for publications -- it can get really toxic!

There is a real and more interesting discussion to be had about whether academics are sufficiently trained in data viz principles to best communicate their data as objectively as possible (avoiding "misleading techniques"), but I don't think that's what people mean by "publication ready".

**Reply by teunbrand** [reply]
> seeing predatory tactics

Wait what? Clearly I have missed or failed to notice something. Are there some examples that are easy for you to find back? If you prefer not to share this publicly, would you mind shooting me an email?

> whether academics are sufficiently trained in data viz principles

From the perspective of my prior nook in academia the answer would be 'no'. I come from a mixed lab where half the people were doing experiments and collecting data and the other half were analysing data. The experimental collegues just want to slap dynamite plots on everything and the computational folks I'd say have a good sense about their visualisations.

> but I don't think that's what people mean by "publication ready".

Yeah fair point

**Reply by yjunechoe** [reply]
> Are there some examples that are easy for you to find back?

It's a fringe corner of social media and none of them go all-in on the data viz side of things (so no specific example off the top of my head), but there some accounts ran by disgruntled academics (or ex-academics) who's made it their whole personally to talk down on academia 😅. They're "predatory" in the sense that they also typically offer services for transitioning out of academia, writing coaching, AI tools, etc. which I guess are not problematic in and of itself, but the business model very clearly does feed off people's fear and anxiety of "not doing enough" or "being in the wrong place". Take this with a grain of salt though, I just have irrationally strong opinions on this 😆 

> From the perspective of my prior nook in academia the answer would be 'no'

Sadly, I also agree! Except we don't have this "experimentalist" vs. "analyst" split -- my field mostly publishes solo-author work, or co-authored with advisor if you're a student. Because a researcher has to wear _all_ the hats (theorist, methodologist, statistician, data analyst, etc.) they inevitably miss out on some training, data viz being one of the more often overlooked one (unsurprisingly, it's is probably also the least important skill among these in getting something published). Funny to hear about your lab dynamics on this!

**Reply by teunbrand** [reply]
Yeah I guess the dynamics in my previous lab are not really classic 'scientific method'. It usually starts with a hypothesis free experiment (we want to see what happens if we do X), a whole bunch of EDA to find some potentially interesting observations, filter the ones that lead to testable hypotheses, send experimentalists back into the lab to collect new/more data, analyse results. Eventually a story takes shape and we polish that up. Solo publications or with your advisor seems scary.

This dynamic isn't always great. At time you feel like you're polishing a turd, but you've already invested resources in data collection. I'd say an important skill to have is to quickly decide if data quality is sufficient to work with.

At least with fellow co-authors on the PhD/postdoc level you can team up against your advisor when you think they're making the wrong decision :p

> their whole personally to talk down on academia

Oh wait, yes, I've come across these people. I haven't had any interactions with them but it does feel kinda icky if your deal is talking people out of their passions.

**Reply by EvaMaeRey** [reply]
>  holier-than-thou and generate FOMO over the latest fad in styling your graph for publications -- it can get really toxic!

I've put together a little package, ggjudge also lets you judge code and text output and lean into the  'holier than thou'! 😬  @yjunechoe I think you're right. There's a lot of side by sides that are 'wrong/right", "tired/inspired", and Drake meme (I know a meme!), that does feel super unnuanced and judgy.  

https://github.com/EvaMaeRey/ggjudge  (gg* here meaning 'returns a ggplot object')

But if there's follow up, and oppenness to disagreement maybe it's okay?  the ggjudge package was inspiredClaus Wilke's Fundamentals book  (and mostly wraps cowplot).    

In Claus' book many plots featured a higher level, un-nuanced judgement.  I feel like those made the book *really* easy to digest... You could dig in further if you didn't really know what 'ugly' or 'bad' judgement was getting at.  But if you did have a strong sense of the principle he's likely to be speaking to, then you could skip or skim the section.  And the judgements also reorient you to the chapter/section when returning back to them.   And judgements might invite discourse more than a nuanced, air-tight argument. 

 I haven't used as much as I'd anticipated - I'm addicted to nuance!) But I did use for {ind2cat} project, which has some publication ready motivations:

![image](https://github.com/user-attachments/assets/d54b76ea-1d42-4a3b-acd3-d9654108cb27)

(Sad because, reader has to do visual search)

![image](https://github.com/user-attachments/assets/f0520a5f-c344-47dd-ae7b-3776bc332315)

(Bad because you can't interpret w/ only 0 1 strip labels)

![image](https://github.com/user-attachments/assets/e5fb07ec-fe56-40fe-8891-35eb8dccc56b)

Tedious because you need to write 'survived' three times. (this is the new offering of ggjudge, beyond what cowplot does!  I feel like judging syntax might be more of a service for generating discussion than judging plots, just because judging plots is so saturated already)

**Reply by mjskay** [reply]
Love this! Always felt that was a nice feature of Claus' book :).

**Reply by EvaMaeRey** [reply]
It's really easy to find your way around his book!

**Reply by EvaMaeRey** [reply]
The labels for hypothetical helper scale_*_logical() proposed for color and fill in #36 might do something like ind2cat::ind_recode().

**Comment by EvaMaeRey** [comment]
Adheres to organization's style guide could be one answer.   But this isn't what the tidyplot is after I think, which is trying to adhere to more generally agreed upon principles.  (but agreement is hard)   But, for example, bar plots where categories are on x axis, I think most people might agree we don't need a vertical grid lines, but ggplot2 defaults to including them.  (I think the ggplot2 defaults are scatter-plot inspired).   I favor a geom_bar(pub = TRUE) kind of approach (or theme_barplot()), which would include barplot theme defaults, which would be more in-line with existing ggplot2 syntax.

**Reply by teunbrand** [reply]
> Adheres to organization's style guide could be one answer.

The style guide of the sender('s institution) (the one who makes the plot) or the receiver (the journal that publishes the article)?

For bar plots in particular, I think it'd also be convenient to bundle the scale expansion so you don't have the 'floating bars' phenomenon by default. Except of course if you have diverging bars.

**Reply by EvaMaeRey** [reply]
I guess I had in mind more than academic publishing.  So self-publishers might have some internal style guide they adhere to (https://urbaninstitute.github.io/graphics-styleguide/) vs. academic publishing where graph crafter and publisher are not same entity.  

I also had 'floating bars' in mind as an example, but I couldn't come up with a term.  so, yes, different scale defaults too for bars might be 'best practice'/publication ready/pretty. And a tick mark seems really unnecessary under the bar, even if on the count axis has one.  I think people just kind of listen to their inner Tufte to try to get things 'publication ready' when it comes to this kind of barchart-type styling.  🙃😬 Does this ink (or spacing) communicate something, or can I get rid of it and nothing is lost?

**Reply by teunbrand** [reply]
I think coords are in charge of expansion as well as gridlines and stuff, so you could just have a `coord_vertical()` and `coord_horizontal()` or something to sync these settings up for bar charts. They even know about the scale limits, so it can detect if the minimum is negative (which would indicate divergent bars).

>  Does this ink (or spacing) communicate something

I'm curious if you have an example about superfluous spacing. Do you mean the spacing underneath the floating bars kind of situation?

**Reply by EvaMaeRey** [reply]
Yeah.  Floating bar spacing is what I was thinking of.  Nothing else is coming to mind.

**Reply by EvaMaeRey** [reply]
And making extra space for bar labels.

**Reply by EvaMaeRey** [reply]
I think it might be interesting to hear from pub-ready extension package author at meetup.  I think ggpubr might be a good candidate since it's been around for quite a while and seems to be really popular.

**Comment by mjskay** [comment]
I semi-regularly teach a PhD seminar called "Visualization for Scientific Communication", which in a way might be considered an attempt to answer the question of what is "publication ready". It has also given me a chance to see standards across a number of fields, as I have had students in the class from a variety of backgrounds. So perhaps I have some useful insight.

The obvious answer is that it varies by field and by lab: there are field-specific norms and even visualization types. There are also venues with more or less strict guidelines. I am fortunate to publish mainly in fields with very few visualization guidelines, which is actually a plus, as in my experience journals with visualization guidelines typically have bad ones.

For me, "publication ready" essentially boils down to: is this a clear, well-designed figure that effectively achieves some specific communication goal, does it integrate well into the rest of the paper, and do I expect it to be well-understood by its intended audience. This standard leads to a bunch of additional implied design guidelines, like:

1. Match effectiveness of visual encodings with importance of data variables. (This is just your typical "learn enough vision science to make important things clear and largely unbiased").
2. Avoid ambiguous or incorrect interpretations. A stricter version of this might be to use encodings that have a strong correspondence between operations in visual space and operations in data space (not always possible, and sometimes must be violated to serve other needs). See [here](https://doi.org/10.1109/TVCG.2014.2346325). An example violation of this is the use of a bar chart on a log scale where 0 is at -Inf and visual ratios of bars are meaningless (in fact, many bar charts have other violations of this; IMO bar charts shouldn't be used nearly as often as they are).
3. Locality is king / eyes beat memory: essentially, don't make readers have to commit things to memory to understand the chart (usually implies using direct labeling instead of legends and trying to make meaningful inferences about data relationships into clear visual patterns).
4. Establish a clear visual hierarchy, with an obvious viewing order and a design that makes it easy to read the chart at different levels of detail.
5. High-quality visual design, which entails a bunch of other things I care about, like consistent use of visual language across figures in a paper, careful use of typography (down to things like keeping font weights well-matched to line weights in a figure), and balancing whitespace and avoiding clutter, etc. I tell students to treat visual attributes like adjectives, and avoid the visual equivalent of sentences like "the very tall building is extremely tall"; that is, don't use three visual attributes where one will do.

This list is non-exhaustive and very much guided by my personal taste. Interestingly, most of the packages that try to make publication ready figures do not satisfy all (or probably most) of my criteria. I think that's fine --- just as people have different writing styles, they have different visual communication styles, so one person's "publication ready" is not necessarily another's. 

For example, @teunbrand and @yjunechoe have both mentioned data journalism-style annotation as being inappropriate in academic contexts. I tend to disagree: I have observed that most academics' figures would benefit from more annotation. In my lab (and in the class I teach) I push students to thoughtfully annotate figures to help guide readers in interpretation. 

The usual pushback against annotation is that it hurts objectivity, perhaps based on a view that visualizations are somehow more objective than text. I do not think this is correct, for a few reasons: (1) if you do not design a figure with a deliberate reading order and visual hierarchy (often aided by text), the reading order will be an accident of the relative salience of elements in the visualization, and likely not a reading order that makes it easy to understand; (2) if the assertion is that text generally is less objective than a visualization, it is odd that scientific papers are mostly text. Rather, it is possible (in fact, easy) to make either text or visualization misleading, and there are very few cases where "objective" visualizations exist. On the other hand, you can combine both together to communicate very effectively.

**Reply by teunbrand** [reply]
As you rightly point out, these criteria signal a lot more thoughfulness about figures than you'd get 'out of the box' with packages that advertise 'publication ready' figures.

> being inappropriate in academic contexts.

Well, I communicated what was impressed upon me. I'm not necessarily an advocate of this position. You do have a mixed medium in an academic article and besides annotations in the figure, you can allocate some space in the main text for storytelling, figure description for extra context or methods for procedural details. This is just to say we shouldn't annotate small novels in figures, but that doesn't make annotations bad.

**Reply by yjunechoe** [reply]
> Well, I communicated what was impressed upon me. I'm not necessarily an advocate of this position.

Echoing this sentiment/clarification too!

One more thing I'd like to add: for me personally, this all feels more like a sociological thing (as in, it's not just about *standards* varying between fields, but *attitudes* towards data viz as a whole too) -- sure, I'd be interested to know how to do academic data viz *well*, whatever that means, but the problem for me is that it's simply not talked about in my field (developmental psychology / language acquisition). I can't expect to get any guidance from my advisors nor the reviewers nor the editors. I could try to self-study and experiment, but tbh the risks and costs outweigh the benefits. So from a practical pov, I've been told to tone down on any viz techniques that deviates from the default / status quo in my field. All that combined, the impression I get as a grad student is that this whole "we don't talk about data viz" situation is actually a very intentional arrangement, because as a field we lack the will/resources/expertise to meaningfully reform our own data viz practices. As with any advancement in any field, _someone_ will need to take the lead and just do it (ex: teach a course on it!), but my impression is that it'll be an uphill battle of convincing people that it's important and will remain a thankless job for a long time to come.

**Reply by teunbrand** [reply]
>  keeping font weights well-matched to line weights in a figure

Also I was intrigued by this. Of course these shouldn't diverge too much, but are there any rules (of thumb)?

**Reply by mjskay** [reply]
> > keeping font weights well-matched to line weights in a figure
>
> Also I was intrigued by this. Of course these shouldn't diverge too much, but are there any rules (of thumb)?

Rules are hard to come by, but I can tell you my opinion ;). I think it's helpful to consider [typographic color](https://en.wikipedia.org/wiki/Type_color) --- which annoyingly doesn't mean "color" but general texture and visual weight. If you can create an even texture it's easier to bump up / emphasize the things you want to emphasize without needing to do the visual equivalent of shouting. One way to judge this is to squint at a visualization / unfocus your eyes so it goes blurry --- this helps you see what regions have a similar level of salience.

If I want two things to be at around the same level of a visual hierarchy, one thing I might do is match their (typographic) "color". For example, one could try to match the font weight of axis labels and the stroke width of the axis. The match may not be mathematically exact, since a long, unbroken axis line at a given stroke width might have a different visual weight than text set in a font with strokes having the same (numerical) width. This is where fonts with a variety of weights (or even better, [variable fonts](https://fonts.adobe.com/fonts/source-sans-3-variable)) can be handy.

As an example, here's a figure from the ggdist paper which I'd like to think ain't too bad:

![image](https://github.com/user-attachments/assets/ac8352e4-96b2-4655-b037-af3f2b8858b8)

If you squint, the top level of the hierarchy (the most salient elements) should be (roughly) the labels of each step ("(A) layer", "(B) stat_slab", ...) and with/just below that, the arrows showing the flow across stages, which have a similar visual weight. This establishes a high-level structure to the diagram, under which you have the lighter-weight text describing each stage and the table for each stage.

**Reply by EvaMaeRey** [reply]
That is a beautiful figure (schematic?) made w/ ggplot2?

**Reply by mjskay** [reply]
Thanks! :)

Unfortunately not made with ggplot2 (except the base figure at the far right). The rest was done in Illustrator. The level of typographic detail I (perhaps unhealthily) obsess over would be hard to do in ggplot2 --- there's a mixture of ~7 weights of just Source Sans Variable in that diagram, and several manual adjustments to kerning, tracking, and baseline shifts.

Though it occurs to me it might be fun to try to make the base table in R as well... I don't know enough about table-generating packages in R to know if it's possible, but I make enough "explodey tables" like these when describing data processing pipelines that it could be worth automating.

**Comment by EvaMaeRey** [comment]
A self-described cookbook-and-package approach to getting to publication ready is bbplot:  https://bbc.github.io/rcookbook/

I think the package handles high level consistent look and feel considerations w/ theme(), and the cookbook guides through lot of chart-type considerations/adjustments.

**Comment by EvaMaeRey** [comment]
Some {tidyplots} discussion in the wild to check out before April 9th meeting. https://www.linkedin.com/posts/%F0%9F%8E%AF-ming-tommy-tang-40650014_want-to-master-bioinformatics-data-visualization-activity-7311745205475463168-FsuI?utm_source=share&utm_medium=member_desktop&rcm=ACoAAAIFK0wBe0Q89iD-ZEWren9Fjl9ugLy3zB0
